swap_meet/vendor.py

```diff
@@ -1,2 +1,66 @@
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if inventory is None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+    
+    def get_by_category(self, category, ):
+        items = []
+        for item in self.inventory:
+            if item.category == category:
+                items.append(item)
+        return items
+    
+    def swap_items(self, vendor_b, my_item, their_item):
+        if my_item not in self.inventory or their_item not in vendor_b.inventory:
+            return False
+
+        self.inventory.append(their_item)
+        vendor_b.inventory.append(my_item)
+        self.inventory.remove(my_item)
+        vendor_b.inventory.remove(their_item)
+        return True
+
+    def swap_first_item(self, vendor_b):
+        
+        if not self.inventory or not vendor_b.inventory:
+            return False
+            
+        swapped_items = self.swap_items(vendor_b, self.inventory[0], vendor_b.inventory[0])
+            
+        return swapped_items
+
+    def get_best_by_category(self, category):
+        items_match_category = self.get_by_category(category)
+        
+        if not items_match_category:
+            return None
+
+        init_condition= 0
+        for item in items_match_category:
+            if item.condition > init_condition:
+                init_condition = item.condition
+                best_item = item
+        return best_item
+    
+    def swap_best_by_category(self, other, my_priority, their_priority):
+        my_best_item = self.get_best_by_category(their_priority)
+        their_best_item = other.get_best_by_category(my_priority)
+
+        swapped_inventory = self.swap_items(other, my_best_item, their_best_item)
+
+        return swapped_inventory
```

Comment:
real nice and concise. you can also even do 

```python
return self.swap_items(other, my_best_item, their_best_item)
```

instead of saving the return value from swap_items() into a variable and then returning the variable